src/fhirtypes/common.ts

```diff
@@ -54,94 +54,367 @@ export function setPropertyOnDefinitionInstance(
 ): void {
   const instanceSD = instance.getOwnStructureDefinition(fisher);
   const { assignedValue, pathParts } = instanceSD.validateValueAtPath(path, value, fisher);
-  setImpliedPropertiesOnInstance(instance, instanceSD, [path], fisher);
+  const knownSlices = determineKnownSlices(instanceSD, new Map([[path, { pathParts }]]), fisher);
+  setImpliedPropertiesOnInstance(instance, instanceSD, [path], [], fisher, knownSlices);
   setPropertyOnInstance(instance, pathParts, assignedValue, fisher);
 }
 
+/**
+ * Adds placeholder elements with slice names to array elements on instance definition.
+ * The placeholder elements to add are based on the paths in the rule map.
+ * @param {StructureDefinition | ElementDefinition | InstanceDefinition} instanceDef - Instance to create slices on
+ * @param {StructureDefinition} instanceOfStructureDefinition - Structure definition for instanceDef
+ * @param {Map<string, { pathParts: PathPart[] }>} ruleMap - Contains the paths used in assignment rules on the instance
+ * @param {Fishable} fisher - A fishable implementation for finding definitions and metadata
+ * Returns the path of the slice including the slice name and the quantity (the minimum or the greatest index of the
+ * slice that is used in a rule) of the slice.
+ */
+export function createUsefulSlices(
+  instanceDef: StructureDefinition | ElementDefinition | InstanceDefinition,
+  instanceOfStructureDefinition: StructureDefinition,
+  ruleMap: Map<string, { pathParts: PathPart[] }>,
+  fisher: Fishable
+): Map<string, number> {
+  const knownSlices = new Map<string, number>();
+  ruleMap.forEach(({ pathParts }, path) => {
+    const nonNumericPath = path.replace(/\[[-+]?\d+\]/g, '');
```

Comment:
We agree that this is weird and maybe not something we should support, but we think since the regex doesn't cause any new issues it's probably best for now to leave it as is and deal with this as a whole later.